storey/targets.py

```diff
@@ -124,14 +127,18 @@ def _init_partition_col_indices(self):
         self._partition_col_indices = []
         self._non_partition_columns = self._columns
 
+        self._non_partition_column_types = self._column_types
         if self._partition_cols is not None and self._columns is not None:
             self._non_partition_columns = []
+            self._non_partition_column_types = []
             for index, col in enumerate(self._columns):
                 if col in self._partition_cols:
                     self._partition_col_to_index[col] = index
                     self._partition_col_indices.append(index)
                 else:
                     self._non_partition_columns.append(col)
+                    if self._column_types:
+                        self._non_partition_column_types.append(self._column_types[index])
```

Comment:
do you want to append only if the type doesn't exists in the list already?

Reply:
Hm, no? That wouldn't make sense :)